src/main/java/ch/njol/skript/hooks/economy/expressions/ExprBalance.java

```diff
@@ -69,44 +61,37 @@ protected String getPropertyName() {
 	
 	@Override
 	@Nullable
-	public Class<?>[] acceptChange(final ChangeMode mode) {
-		if (mode == ChangeMode.REMOVE_ALL)
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.REMOVE_ALL || mode == ChangeMode.DELETE)
```

Comment:
I think it would be okay to keep the DELETE changer supported. While `delete` isn't the best sounding like you mentioned, the DELETE changer also supports `clear` and I think `clear player's balance` is fine.